[BUG] test_from_utc_timestamp failed on Cloudera Env when TZ is Iran

**Describe the bug**
test_from_utc_timestamp failed on Cloudera Env when TZ is Iran
Navin found:
```
the timezone Iran has transition rules in my JVM, 
we should not recommend it as non-UTC timezone for testing as this means it supports DST. 
```
`test_from_utc_timestamp` can pass when TZ is `Asia/Shanghai`, but failed on `Iran` TZ.
Note: 
This error happends on Cloudera when global TZ is UTC. The TZ here refers to the TZ parameter in `from_utc_timestamp` function.

**Steps/Code to reproduce bug**
The error on Cloudera CI
```
[2023-12-28T16:15:37.579Z] =================================== FAILURES ===================================

[2023-12-28T16:15:37.579Z] ___________________ test_from_utc_timestamp[Timestamp-Iran] ____________________

[2023-12-28T16:15:37.579Z] 

[2023-12-28T16:15:37.579Z] data_gen = Timestamp, time_zone = 'Iran'

[2023-12-28T16:15:37.579Z] 

[2023-12-28T16:15:37.579Z]     @pytest.mark.parametrize('time_zone', ["Asia/Shanghai", "Iran", "UTC", "UTC+0", "UTC-0", "GMT", "GMT+0", "GMT-0"], ids=idfn)

[2023-12-28T16:15:37.579Z]     @pytest.mark.parametrize('data_gen', [timestamp_gen], ids=idfn)

[2023-12-28T16:15:37.579Z]     @tz_sensitive_test

[2023-12-28T16:15:37.579Z]     @allow_non_gpu(*non_utc_allow)

[2023-12-28T16:15:37.579Z]     def test_from_utc_timestamp(data_gen, time_zone):

[2023-12-28T16:15:37.579Z] >       assert_gpu_and_cpu_are_equal_collect(

[2023-12-28T16:15:37.579Z]             lambda spark: unary_op_df(spark, data_gen).select(f.from_utc_timestamp(f.col('a'), time_zone)))

[2023-12-28T16:15:37.579Z] 

[2023-12-28T16:15:37.579Z] ../../src/main/python/date_time_test.py:295: 

[2023-12-28T16:15:37.579Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

[2023-12-28T16:15:37.579Z] ../../src/main/python/asserts.py:595: in assert_gpu_and_cpu_are_equal_collect

[2023-12-28T16:15:37.579Z]     _assert_gpu_and_cpu_are_equal(func, 'COLLECT', conf=conf, is_cpu_first=is_cpu_first, result_canonicalize_func_before_compare=result_canonicalize_func_before_compare)

[2023-12-28T16:15:37.579Z] ../../src/main/python/asserts.py:503: in _assert_gpu_and_cpu_are_equal

[2023-12-28T16:15:37.579Z]     from_gpu = run_on_gpu()

[2023-12-28T16:15:37.579Z] ../../src/main/python/asserts.py:496: in run_on_gpu

[2023-12-28T16:15:37.579Z]     from_gpu = with_gpu_session(bring_back, conf=conf)

[2023-12-28T16:15:37.579Z] ../../src/main/python/spark_session.py:164: in with_gpu_session

[2023-12-28T16:15:37.579Z]     return with_spark_session(func, conf=copy)

[2023-12-28T16:15:37.579Z] /usr/local/lib/python3.8/contextlib.py:75: in inner

[2023-12-28T16:15:37.579Z]     return func(*args, **kwds)

[2023-12-28T16:15:37.579Z] ../../src/main/python/spark_session.py:131: in with_spark_session

[2023-12-28T16:15:37.579Z]     ret = func(_spark)

[2023-12-28T16:15:37.579Z] ../../src/main/python/asserts.py:205: in <lambda>

[2023-12-28T16:15:37.579Z]     bring_back = lambda spark: limit_func(spark).collect()

[2023-12-28T16:15:37.579Z] /opt/cloudera/parcels/SPARK3-3.3.0.3.3.7180.0-274-1.p0.31212967/lib/spark3/python/lib/pyspark.zip/pyspark/sql/dataframe.py:817: in collect

[2023-12-28T16:15:37.579Z]     sock_info = self._jdf.collectToPython()

[2023-12-28T16:15:37.579Z] /opt/cloudera/parcels/SPARK3-3.3.0.3.3.7180.0-274-1.p0.31212967/lib/spark3/python/lib/py4j-0.10.9.5-src.zip/py4j/java_gateway.py:1321: in __call__

[2023-12-28T16:15:37.579Z]     return_value = get_return_value(

[2023-12-28T16:15:37.579Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

[2023-12-28T16:15:37.579Z] 

[2023-12-28T16:15:37.579Z] a = ('xro21350', <py4j.clientserver.JavaClient object at 0x7f484578e520>, 'o21349', 'collectToPython')

[2023-12-28T16:15:37.579Z] kw = {}, converted = IllegalArgumentException()

[2023-12-28T16:15:37.579Z] 

[2023-12-28T16:15:37.579Z]     def deco(*a: Any, **kw: Any) -> Any:

[2023-12-28T16:15:37.579Z]         try:

[2023-12-28T16:15:37.579Z]             return f(*a, **kw)

[2023-12-28T16:15:37.579Z]         except Py4JJavaError as e:

[2023-12-28T16:15:37.579Z]             converted = convert_exception(e.java_exception)

[2023-12-28T16:15:37.579Z]             if not isinstance(converted, UnknownException):

[2023-12-28T16:15:37.579Z]                 # Hide where the exception came from that shows a non-Pythonic

[2023-12-28T16:15:37.579Z]                 # JVM exception message.

[2023-12-28T16:15:37.579Z] >               raise converted from None

[2023-12-28T16:15:37.579Z] E               pyspark.sql.utils.IllegalArgumentException: Part of the plan is not columnar class org.apache.spark.sql.execution.ProjectExec

[2023-12-28T16:15:37.579Z] E               Project [from_utc_timestamp(a#1873, Iran) AS from_utc_timestamp(a, Iran)#1875]

[2023-12-28T16:15:37.579Z] E               +- Scan ExistingRDD[a#1873]

[2023-12-28T16:15:37.579Z] 

[2023-12-28T16:15:37.579Z] /opt/cloudera/parcels/SPARK3-3.3.0.3.3.7180.0-274-1.p0.31212967/lib/spark3/python/lib/pyspark.zip/pyspark/sql/utils.py:196: 
```

--- a/integration_tests/README.md
+++ b/integration_tests/README.md
@@ -350,10 +350,10 @@ But only a small amout of cases are verifing with non-UTC time zone in the pre-m
 When adding cases, should also check non-UTC is working besides the default UTC time zone.
 Please test the following time zones:
 ```shell
-$ TZ=Iran ./integration_tests/run_pyspark_from_build.sh
+$ TZ=Asia/Shanghai ./integration_tests/run_pyspark_from_build.sh
 $ TZ=America/Los_Angeles ./integration_tests/run_pyspark_from_build.sh
 ```
-`Iran` is non-DST(Daylight Savings Time) time zone and `America/Los_Angeles` is DST time zone.
+`Asia/Shanghai` is non-DST(Daylight Savings Time) time zone and `America/Los_Angeles` is DST time zone.
 
 If the new added cases failed with non-UTC, then should allow the operator(does not support non-UTC) fallback,
 For example, add the following annotation to the case:

--- a/integration_tests/src/main/python/date_time_test.py
+++ b/integration_tests/src/main/python/date_time_test.py
@@ -287,7 +287,7 @@ def test_unsupported_fallback_to_unix_timestamp(data_gen):
         "to_unix_timestamp(a, b)"),
         "ToUnixTimestamp")
 
-@pytest.mark.parametrize('time_zone', ["Asia/Shanghai", "Iran", "UTC", "UTC+0", "UTC-0", "GMT", "GMT+0", "GMT-0"], ids=idfn)
+@pytest.mark.parametrize('time_zone', ["Asia/Shanghai", "UTC", "UTC+0", "UTC-0", "GMT", "GMT+0", "GMT-0"], ids=idfn)
 @pytest.mark.parametrize('data_gen', [timestamp_gen], ids=idfn)
 @tz_sensitive_test
 @allow_non_gpu(*non_utc_allow)

--- a/tests/src/test/scala/com/nvidia/spark/rapids/timezone/TimeZonePerfSuite.scala
+++ b/tests/src/test/scala/com/nvidia/spark/rapids/timezone/TimeZonePerfSuite.scala
@@ -34,7 +34,7 @@ import org.apache.spark.sql.types._
  * A simple test performance framework for non-UTC features.
  * Usage:
  *
- *   argLine="-DTZs=Iran,Asia/Shanghai -DenableTimeZonePerf=true" \
+ *   argLine="-DTZs=Asia/Shanghai,Japan -DenableTimeZonePerf=true" \
  *   mvn test -Dbuildver=311 -DwildcardSuites=com.nvidia.spark.rapids.timezone.TimeZonePerfSuite
  * Note:
  *   Generate a Parquet file with 6 columns:
@@ -51,7 +51,7 @@ import org.apache.spark.sql.types._
 class TimeZonePerfSuite extends SparkQueryCompareTestSuite with BeforeAndAfterAll {
   // perf test is disabled by default since it's a long running time in UT.
   private val enablePerfTest = java.lang.Boolean.getBoolean("enableTimeZonePerf")
-  private val timeZoneStrings = System.getProperty("TZs", "Iran")
+  private val timeZoneStrings = System.getProperty("TZs", "Asia/Shanghai")
 
   // rows for perf test
   private val numRows: Long = 1024L * 1024L * 10L

